pkg/cmd/server/api/validation/master.go

```diff
@@ -654,7 +654,7 @@ func ValidateAPIServerExtendedArguments(config api.ExtendedArguments, fldPath *f
 }
 
 func ValidateControllerExtendedArguments(config api.ExtendedArguments, fldPath *field.Path) field.ErrorList {
-	return ValidateExtendedArguments(config, controlleroptions.NewCMServer().AddFlags, fldPath)
```

Comment:
I did something slightly different in https://github.com/sttts/origin/pull/23/commits/34ed4601ed601aa35b9b1adb68dcfcbe948bde70. I think my way is ok for validation, but I think your way is better in the long run. cc @deads2k